flink-end-to-end-tests/flink-datastream-allround-test/src/main/java/org/apache/flink/streaming/tests/DataStreamAllroundTestProgram.java

```diff
@@ -172,6 +179,30 @@ public void apply(Integer integer, TimeWindow window, Iterable<Event> input, Col
 			in -> (Integer) in.get("key"),
 			"Avro (Generic)");
 
+		// test operator input serialization that goes through Kryo, without the type registered
+		eventStream3 = testSpecificOperatorInputTypeSerialization(
+			eventStream3,
+			in -> new GenericKryoEvent(in.getKey(), in.getEventTime(), in.getSequenceNumber(), in.getPayload()),
+			in -> new Event(in.getKey(), in.getEventTime(), in.getSequenceNumber(), in.getPayload()),
+			GenericKryoEvent::getKey,
+			"Kryo (Generic)");
```

Comment:
Should we also pass the expected serializer class to this method or some validation lambda that takes the serializer such that we can perform some validation to make sure we are testing the right thing?